### PR TITLE
[RFR] Two players

### PR DESCRIPTION
**Pull request by HALLERPierre**
# TODO
- [x] Add join game screen
- [x] Add assign team API
- [x] Add instructions

# Result
2-p

### CODE REVIEW

File: api/tests.js/game/game.test.js [hunk 1]
```diff
@@ -64,6 +64,7 @@ describe("Game functions", () => {
 
       const destinations = getAvailablesDestinations({ ...game, selectedCube });
       expect(destinations).toEqual(exepectedDestinations);
+      expect(destinations.length).toEqual(exepectedDestinations.length);
```
Comment (zyhou) [inline api/tests.js/game/game.test.js]: `toBe` instead of `toEqual`

File: quixo-react-native/game/actions.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  SELECT_CUBE_REPLY,
+  FETCH_TEAM_REPLY
+} from "../game/constants";
+import { CIRCLE_VALUE } from "../constants/game";
+
+const storeMyTeam = async (team, id) =>
+  await AsyncStorage.setItem(`@Quixo-game-${id}`, team.toString());
+
+const getTeamFromStorage = async id =>
+  await AsyncStorage.getItem(`@Quixo-game-${id}`);
```
Comment (zyhou) [inline quixo-react-native/game/actions.js]: parse the team result here

File: quixo-react-native/game/actions.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+export const refreshGame = (id, isPlaying, dispatch) => () => {
+  const refreshGameCall = async () => {
+    const payload = await getExistingGame(id);
+    dispatch({ type: UPDATE_GAME, payload });
+  };
+  let intervalId;
+  if (!isPlaying && id) {
+    intervalId = setInterval(refreshGameCall, 1000);
+  }
+  return () => clearInterval(intervalId);
```
Comment (zyhou) [inline quixo-react-native/game/actions.js:99]: +1

File: quixo-react-native/screens/GameScreen.js [hunk 4]
```diff
@@ -34,68 +72,69 @@ const isInMovables = movables => ({ x, y }) =>
...
+const isMyTurn = (myTeam, currentPlayer) => myTeam === currentPlayer;
 
-const fetchMovables = (id, dispatch) => () => {
-  const fetchMovablesCall = async () => {
-    const payload = await getMovables(id);
-    dispatch({ type: FETCH_MOVABLES, payload });
-  };
-  if (id) {
-    fetchMovablesCall();
+const getInstructions = (team, isPlaying) => {
```
Comment (zyhou) [inline quixo-react-native/screens/GameScreen.js]: Move the component in another file. Rename to `<Instructions />`

File: quixo-react-native/screens/GameScreen.js [hunk 5]
```diff
@@ -34,68 +72,69 @@ const isInMovables = movables => ({ x, y }) =>
...
-  const id = navigation.getParam("id", null) || game.id;
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const { game, movables, myTeam } = state;
+  const id = game.id || navigation.getParam("id", null);
+  const isPlaying = isMyTurn(myTeam, game.currentPlayer);
+
+  useEffect(refreshGame(id, isPlaying, dispatch), null);
   useEffect(fetchGame(id, dispatch), []);
   useEffect(fetchMovables(id, dispatch), [game.currentPlayer]);
+  useEffect(fetchMyTeam(id, dispatch), [game.id]);
```
Comment (zyhou) [inline quixo-react-native/screens/GameScreen.js]: why not `[id]` ?

File: quixo-react-native/screens/GameScreen.js [hunk 6]
```diff
@@ -34,68 +72,69 @@ const isInMovables = movables => ({ x, y }) =>
...
+      {getInstructions(myTeam, isPlaying)}
+      <View style={styles.board}>
+        {game.board.map((row, x) => (
+          <View key={`row-${x}`} style={styles.row}>
+            {row.map((value, y) => (
+              <Cube
+                key={`cube-${x}-${y}`}
+                isMovable={isMovable({ x, y })}
+                isSelected={isSelected({ x, y })}
+                handlePress={() => handlePressCube({ x, y })}
```
Comment (zyhou) [inline quixo-react-native/screens/GameScreen.js]: remove this func

**Review by zyhou — CHANGES_REQUESTED** [review]
Can you test the component and the `game/actions.js` file ?

File: quixo-react-native/screens/GameScreen.js [hunk 7]
```diff
@@ -34,68 +72,69 @@ const isInMovables = movables => ({ x, y }) =>
...
-    dispatch({ type: FETCH_MOVABLES, payload });
-  };
-  if (id) {
-    fetchMovablesCall();
+const getInstructions = (team, isPlaying) => {
+  const isSpectator = team === NEUTRAL_VALUE;
+  if (isSpectator) {
+    return (
+      <View style={styles.instructions}>
+        <Text style={styles.instructionsText}>You're a spectator </Text>
```
Comment (Kmaschta) [inline quixo-react-native/screens/GameScreen.js]: Apostrophe are breaking GitHub diff. Nothing to really worry about, but it's nice if you wrap "You're" with a template literal next time. `{"You're"}`

**Review by zyhou — APPROVED** [review]

**Pull request merged**